src/seedu/addressbook/commands/Command.java

```diff
@@ -16,6 +16,15 @@
     protected List<? extends ReadOnlyPerson> relevantPersons;
     private int targetIndex = -1;
 
+    /**
+     * Use this to decide if the addressbook data needs to be saved to disk
+     *
+     * @return true if this command mutates the data
+     */
+    public boolean isMutating() {
+        return true;
```

Comment:
Having a default implementation reduces the amount of code needed, but might be more unintuitive. Maybe you can make the method abstract to force the child classes to declare whether it is a mutating command or not.